### ISSUE TITLE
Fix S4275 false positives

### ISSUE THREAD

**Issue by costin-zaharia-sonarsource** [SonarSource/sonar-dotnet#4339]
### Description

There are multiple FP raised by [S4275](https://rules.sonarsource.com/csharp/RSPEC-4275).
sonar-dotnet version: 8.22

### Repro steps

```
public class TestCases
    {
        bool pause;
        public bool Pause
        {
            get => pause;
            set => pause |= value; // Noncompliant - FP
        }

        private int textBufferUndoHistory;
        public int TextBufferUndoHistory
        {
            get // Noncompliant - FP
            {
                return textBufferUndoHistory = GetValue();
            }
        }
        private static int GetValue() => 1;

        private int[] attributes;
        public int[] Attributes
        {
            set { value.CopyTo(attributes, 0); } // Noncompliant - FP
        }

        private const string PREFIX = "pre";
        private string m_prefix;
        public string Prefix
        {
            get { return m_prefix; } // Noncompliant - FP
            set { m_prefix = value; } // Noncompliant - FP
        }
    }

    public class ContainsConstraint
    {
        private bool _ignoreCase;

        public ContainsConstraint IgnoreCase
        {
            // Refactor this getter so that it actually refers to the field '_ignoreCase'.
            // _ignoreCase has a different type than the property return type
            get // Noncompliant - FP
            {
                _ignoreCase = true;
                return this;
            }
        }
    }

    public class Base
    {
        protected long writeLockTimeout;
    }

    public class SubClass : Base
    {
        public static long WRITE_LOCK_TIMEOUT = 1000;
        public long WriteLockTimeout
        {
            get
            {
                return writeLockTimeout; // Noncompliant - FP
            }
            set
            {
                this.writeLockTimeout = value; // Noncompliant - FP
            }
        }
    }
```

**Comment by mary-georgiou-sonarsource** [comment]
FP related to setting a property with a method was not fixed.
There's an open issue (#5259) for it to be fixed in a next PR.
```csharp
        public int[] Attributes
        {
            set { value.CopyTo(attributes, 0); } // Noncompliant - FP
        }
  ```
